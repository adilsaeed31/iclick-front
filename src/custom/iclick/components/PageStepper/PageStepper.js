import React, { Component } from "react";
import PropTypes from "prop-types";
import Button from "@reactioncommerce/components/Button/v1";

export default class PageStepper extends Component {
  static propTypes = {
    pageInfo: PropTypes.shape({
      hasNextPage: PropTypes.bool,
      hasPreviousPage: PropTypes.bool,
      loadNextPage: PropTypes.func,
      loadPreviousPage: PropTypes.func
    }).isRequired,
    theme: PropTypes.object
  }

  handleNextClick = () => {
    const { pageInfo } = this.props;

    if (typeof window !== "undefined" && typeof window.scrollTo === "function") {
      window.scrollTo({ top: 0, behavior: "smooth" });
    }

    pageInfo.loadNextPage();
  }

  handlePreviousClick = () => {
    const { pageInfo } = this.props;

    if (typeof window !== "undefined" && typeof window.scrollTo === "function") {
      window.scrollTo({ top: 0, behavior: "smooth" });
    }

    pageInfo.loadPreviousPage();
  }

  render() {
    const { pageInfo } = this.props;

    return (

      <ul className="pagination">
         <li className="page-item">
            <a className="page-link page-link-btn" href="#">
              {pageInfo.hasPreviousPage && <a className="icon-angle-left" onClick={this.handlePreviousClick}></a>}
            </a>
          </li>
          <li className="page-item active">
            <a className="page-link" href="#">1 <span className="sr-only">(current)</span></a>
          </li>
          <li className="page-item"><span className="page-link">...</span></li>
          <li className="page-item">
            <a className="page-link page-link-btn" href="#">
              {pageInfo.hasNextPage && <a className="icon-angle-right" onClick={this.handleNextClick}></a>}
            </a>
          </li>
      </ul> 
    );
  }
}
