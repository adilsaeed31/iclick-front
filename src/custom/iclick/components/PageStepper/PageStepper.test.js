import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Button from "@reactioncommerce/components/Button/v1";
import PageStepper from "./PageStepper";

test("shows next button, hides previous button", () => {
  const pageInfo = {
    hasNextPage: true,
    hasPreviousPage: false
  };

  const component = renderer.create(<PageStepper pageInfo={pageInfo} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("hides next button, shows previous button", () => {
  const pageInfo = {
    hasNextPage: false,
    hasPreviousPage: true
  };

  const component = renderer.create(<PageStepper pageInfo={pageInfo} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("shows both next and previous buttons", () => {
  const pageInfo = {
    hasNextPage: true,
    hasPreviousPage: true
  };

  const component = renderer.create(<PageStepper pageInfo={pageInfo} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("click next button", () => {
  const pageInfo = {
    hasNextPage: true,
    loadNextPage: jest.fn()
  };

  const wrapper = mount(<PageStepper pageInfo={pageInfo} />);

  const buttonElement = wrapper.find(Button);
  expect(buttonElement.length).toBe(1);
  buttonElement.simulate("click");
  expect(pageInfo.loadNextPage).toHaveBeenCalled();
});

test("click previous button", () => {
  const pageInfo = {
    hasPreviousPage: true,
    loadPreviousPage: jest.fn()
  };

  const wrapper = mount(<PageStepper pageInfo={pageInfo} />);

  const buttonElement = wrapper.find(Button);
  expect(buttonElement.length).toBe(1);
  buttonElement.simulate("click");
  expect(pageInfo.loadPreviousPage).toHaveBeenCalled();
});
