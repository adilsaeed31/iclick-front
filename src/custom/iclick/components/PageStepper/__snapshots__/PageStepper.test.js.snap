// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides next button, shows previous button 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #5e7480;
  border-color: #5e7480;
  border-style: solid;
  border-width: 1px;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Source Sans Pro","Helvetica Neue",Helvetica,sans-serif;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  min-width: 100px;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 10px;
  position: relative;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0:hover {
  background-color: #5d8fa9;
  border-color: #5d8fa9;
  color: #ffffff;
}

.c0:active {
  background-color: #3b5c70;
  border-color: #3b5c70;
  color: #ffffff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  padding-left: 20px;
  -webkit-transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
  transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
}

.c1 svg path,
.c1 svg rect {
  fill: #ffffff;
}

<div
  className="MuiGrid-container-2 MuiGrid-justify-xs-space-between-21 SkPageStepper-root-1"
>
  <div
    className="MuiGrid-item-3"
  >
    <div
      className="c0"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div>
        Previous
      </div>
      <div
        className="c1"
        style={
          Object {
            "opacity": 0,
            "paddingLeft": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiGrid-item-3"
  />
</div>
`;

exports[`shows both next and previous buttons 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #5e7480;
  border-color: #5e7480;
  border-style: solid;
  border-width: 1px;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Source Sans Pro","Helvetica Neue",Helvetica,sans-serif;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  min-width: 100px;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 10px;
  position: relative;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0:hover {
  background-color: #5d8fa9;
  border-color: #5d8fa9;
  color: #ffffff;
}

.c0:active {
  background-color: #3b5c70;
  border-color: #3b5c70;
  color: #ffffff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  padding-left: 20px;
  -webkit-transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
  transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
}

.c1 svg path,
.c1 svg rect {
  fill: #ffffff;
}

<div
  className="MuiGrid-container-2 MuiGrid-justify-xs-space-between-21 SkPageStepper-root-1"
>
  <div
    className="MuiGrid-item-3"
  >
    <div
      className="c0"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div>
        Previous
      </div>
      <div
        className="c1"
        style={
          Object {
            "opacity": 0,
            "paddingLeft": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiGrid-item-3"
  >
    <div
      className="c0"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div>
        Next
      </div>
      <div
        className="c1"
        style={
          Object {
            "opacity": 0,
            "paddingLeft": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`shows next button, hides previous button 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #5e7480;
  border-color: #5e7480;
  border-style: solid;
  border-width: 1px;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Source Sans Pro","Helvetica Neue",Helvetica,sans-serif;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  min-width: 100px;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 10px;
  position: relative;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0:hover {
  background-color: #5d8fa9;
  border-color: #5d8fa9;
  color: #ffffff;
}

.c0:active {
  background-color: #3b5c70;
  border-color: #3b5c70;
  color: #ffffff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  padding-left: 20px;
  -webkit-transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
  transition: width 0.2s ease-out 0s,padding-left 0.2s ease-out 0s,opacity 0.2s ease-out 0.2s;
}

.c1 svg path,
.c1 svg rect {
  fill: #ffffff;
}

<div
  className="MuiGrid-container-2 MuiGrid-justify-xs-space-between-21 SkPageStepper-root-1"
>
  <div
    className="MuiGrid-item-3"
  />
  <div
    className="MuiGrid-item-3"
  >
    <div
      className="c0"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div>
        Next
      </div>
      <div
        className="c1"
        style={
          Object {
            "opacity": 0,
            "paddingLeft": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;
